src/peft/tuners/reft/layer.py

```diff
@@ -0,0 +1,262 @@
+# Copyright 2023-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import warnings
+from typing import Any, List, Optional, Set, Tuple
+
+import torch
+import torch.nn as nn
+
+from peft.tuners.lycoris_utils import LycorisLayer, check_adapters_to_merge
+
+class LowRankRotateLayer(torch.nn.Module):
```

Comment:
Could this be replaced by an `nn.Linear` with `bias=False` and then just apply the orthogonal initialization on its weight?

Alternatively, we could modify the `forward` to take `rotated_base` and `offset` and perform both operations.